### DISCUSSION TITLE
global logging breaks when uvicorn.run(reload=True)

### DISCUSSION THREAD

**Discussion by Lx** [Kludex/uvicorn#1498] [Potential Issue]
I have a script that sets up logging via `logging.basicConfig`, then starts Uvicorn using `uvicorn.run`.

If I **don't** use `reload=True` in the `uvicorn.run` call, logging works as expected.

If I **do** use `reload=True` in the `uvicorn.run` call:
- the `root` logger (belonging to `logging`) gets obliterated (its `id` even changes)
- all custom logging configuration is lost
- all existing loggers revert to printing just their messages to `stderr`.

This problem exists in the latest released version at the time of this discussion being opened: 0.17.6.

Am I okay to raise this as a "real" issue?

---

One further detail in case it helps: the problem exists with or without installation of `watchgod`.

**Comment by Kludex** [comment]
Do you have a repository/gist so I can reproduce that? 🙏

**Reply by Kludex** [reply]
This https://github.com/encode/uvicorn/issues/328 ?

**Reply by Lx** [reply]
Probably the minimal useful example (which I appreciate is still not very useful) would be:

```py
logging.basicConfig(
    level="NOTSET",
    format="[%(name)s] %(message)s",
)
uvicorn.run(
    app="thingo.server:app",
    host="0.0.0.0",  # nosec
    reload=True,  # TODO: work out why "reload=True" breaks logging
)
```

In my scenario, and in case it's relevant, `things.server:app` is a FastAPI app.  I can't offer a repo URL as it's a private project.

**Reply by Lx** [reply]
It doesn't seem to be the same problem as #328, because no exception is thrown and I'm using Python 3.10.4 for this project.
